pkg/controller/job/job_controller.go

```diff
@@ -799,16 +794,12 @@ func (jm *Controller) syncJob(ctx context.Context, key string) (forget bool, rEr
 	// Remove active pods if Job failed.
 	if finishedCondition != nil {
 		deleted, err := jm.deleteActivePods(ctx, &job, activePods)
-		if uncounted == nil {
-			// Legacy behavior: pretend all active pods were successfully removed.
-			deleted = active
-		} else if deleted != active || !satisfiedExpectations {
+		if deleted != active || !satisfiedExpectations {
 			// Can't declare the Job as finished yet, as there might be remaining
 			// pod finalizers or pods that are not in the informer's cache yet.
 			finishedCondition = nil
 		}
 		active -= deleted
-		failed += deleted
```

Comment:
Were we counting pods as failed twice? here and in `trackJobStatusAndRemoveFinalizers`. If so, we probably should have a fix in 1.23 to 1.26.
And a unit test that covers this case in all versions.

Reply:
Did you figure this question?

Reply:
In the legacy path, we were setting `job.Status.Failed = failed`.  I don't see us setting anything like that in trackJobStatusAndRemoveFinalizers.  

I removed this because once I deleted the legacy path, we were no longer using that variable anywhere else.  

In `trackJobStatusAndRemoveFinalizers` we do set  `uncountedStatus.Failed` using the uncounted.  

Here is the relevant code:

```
			ix := getCompletionIndex(pod.Annotations)
			if !uncounted.failed.Has(string(pod.UID)) && (!isIndexed || (ix != unknownCompletionIndex && ix < int(*job.Spec.Completions))) {
				if feature.DefaultFeatureGate.Enabled(features.JobPodFailurePolicy) && job.Spec.PodFailurePolicy != nil {
					_, countFailed, action := matchPodFailurePolicy(job.Spec.PodFailurePolicy, pod)
					if action != nil {
						podFailureCountByPolicyAction[string(*action)] += 1
					}
					if countFailed {
						needsFlush = true
						uncountedStatus.Failed = append(uncountedStatus.Failed, pod.UID)
					}
				} else {
					needsFlush = true
					uncountedStatus.Failed = append(uncountedStatus.Failed, pod.UID)
				}
			}
```